### PR TITLE
rgw_sal_motr : [CORTX-31876] for unversioned/suspended bucket, skip versionId parameter for GET/HEAD object API response

### PR DESCRIPTION
**Pull request by sachitanands**
Problem:
In un-versioned or suspended bucket, version-id field with null value is displayed in response of GET and HEAD object API.
In case of RADOS & AWS, this null version-id field is ignored from above API response.

Solution:
If bucket is un-versioned/suspended and version-id is null then delete the versiond-id value from result.

Signed-off-by: Sachitanand Shelake <sachitanand.shelake@seagate.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sachitanands** [comment]
Triggered build for testing those changes.
custom-ci job : https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6605/ (In-progress)

**Review by shriya-deshmukh — APPROVED** [review]

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -1741,6 +1741,13 @@ int MotrObject::MotrReadOp::prepare(optional_yield y, const DoutPrefixProvider*
   if (rc < 0)
     return rc;
 
+  // In case of un-versioned/suspended case,
+  // GET and HEAD object output should ignore versiodId field.
+  if (!source->have_instance()) {
+    if (ent.key.instance == "null")
```
[the remark this answers is not in the document]
Comment (sachitanands) [reply src/rgw/rgw_sal_motr.cc:1770]: in case of null version, this versionId field itself will be skipped in response

**Comment by sachitanands** [comment]
rebased those changes and retriggered rpm build
https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6638/console

**Comment by sachitanands** [comment]
After rebase, triggered new build job 
ci-job: https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6651/console (In-progress)

**Comment by sachitanands** [comment]
After rebase, triggered new build job 
ci-job: https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6651/console (In-progress)

**Comment by sachitanands** [comment]
> After rebase, triggered new build job ci-job: https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6651/console cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-6651-custom-ci cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-6651-custom-ci cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-6651-custom-ci
> 
> Triggered deployment: https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/6666 (In-progress)

I have tested above PR and captured all the build, deployment results and manual commands tried on this at,
https://jts.seagate.com/browse/CORTX-31876?focusedCommentId=2380445&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2380445

**Pull request merged**